sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/CatalystQl.scala

```diff
@@ -661,6 +664,36 @@ https://cwiki.apache.org/confluence/display/Hive/Enhanced+Aggregation%2C+Cube%2C
   val INTEGRAL = "[+-]?\\d+".r
   val DECIMAL = "[+-]?((\\d+(\\.\\d*)?)|(\\.\\d+))".r
 
+  /**
+   * Flattens the left deep tree with the specified pattern into a list.
+   */
+  private def flattenLeftDeepTree(node: ASTNode, pattern: Regex): Seq[ASTNode] = {
+    val collected = ArrayBuffer[ASTNode]()
+    var rest = node
+    while (rest match {
+      case Token(pattern(), l :: r :: Nil) =>
+        collected += r
+        rest = l
+        true
+      case _ => false
+    }) {}
+    collected += rest
+    collected.reverse
```

Comment:
We want to keep the same order as SQL string,

`A OR B OR C` will become  (OR (OR A B) C), collected will be (C, B, A), we should return (A, B, C)